pkg/queue/queue.go

```diff
@@ -149,6 +169,56 @@ func (cq *ClusterQueue) Pop() *workload.Info {
 	return &w
 }
 
+func (cq *ClusterQueue) RequeueWorkload(wInfo *workload.Info) (bool, error) {
+	switch cq.QueueingStrategy {
+	case kueue.StrictFIFO:
+		return cq.PushIfNotPresent(wInfo), nil
+	case kueue.BestEffortFIFO:
+		if err := cq.AddUnschedulableIfNotPresent(wInfo); err != nil {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+// AddUnschedulableIfNotPresent inserts a workload that cannot be scheduled into
+// the unschedulableWorkloads in ClusterQueue, unless it is already in the queue.
+func (cq *ClusterQueue) AddUnschedulableIfNotPresent(wInfo *workload.Info) error {
+	w := wInfo.Obj
+	if cq.unschedulableWorkloads.get(w) != nil {
+		return fmt.Errorf("workload %v is already present in unschedulableWorkloads", workload.Key(w))
+	}
+
+	if _, exist := cq.heap.Get(wInfo); exist {
+		return fmt.Errorf("workload %v is already present in the clusterqueue heap", workload.Key(w))
+	}
+
+	cq.unschedulableWorkloads.addOrUpdate(wInfo)
+	return nil
+}
+
+// MoveAllToClusterQueue moves all workloads from unschedulableWorkloads to heap.
+func (cq *ClusterQueue) MoveAllToClusterQueue() {
+	switch cq.QueueingStrategy {
+	case kueue.StrictFIFO:
+		return
+	}
+
+	unschedulableWorkloads := make([]*workload.Info, 0, len(cq.unschedulableWorkloads.workloads))
+	for _, pInfo := range cq.unschedulableWorkloads.workloads {
+		unschedulableWorkloads = append(unschedulableWorkloads, pInfo)
+	}
+	cq.moveWorkloadsToClusterQueue(unschedulableWorkloads)
+}
+
+func (cq *ClusterQueue) moveWorkloadsToClusterQueue(workloads []*workload.Info) {
```

Comment:
perhaps unroll this in the function above, doesn't serve too much purpose on its own and the func above is short.

Reply:
Done